src/connection.c

```diff
@@ -2010,8 +2013,14 @@ void connection_destroy_endpoint(ENDPOINT_HANDLE endpoint)
                     (void)memmove(connection->endpoints + i, connection->endpoints + i + 1, sizeof(ENDPOINT_HANDLE) * (connection->endpoint_count - i - 1));
                 }
 
-                new_endpoints = (ENDPOINT_HANDLE*)realloc(connection->endpoints, (connection->endpoint_count - 1) * sizeof(ENDPOINT_HANDLE));
-                if (new_endpoints != NULL)
+                size_t realloc_size = safe_subtract_size_t(connection->endpoint_count, 1);
+                realloc_size = safe_multiply_size_t(realloc_size, sizeof(ENDPOINT_HANDLE));
+                if (realloc_size == SIZE_MAX ||
```

Comment:
this can't be zero because there is a count == 1 check about it which frees the endpoint memory

There is a memmove with int overflow, so I added safe math there as well